packages/app/src/utils/tenderSwapHooks.ts

```diff
@@ -65,6 +66,39 @@ export const useCalculateSwap = (pool: string, tokenFrom: string, amount: BigNum
   return tokens;
 };
 
+export const useSwapWithPermit = (
+  token: string,
+  protocolName: string,
+  owner: string | null | undefined,
+  spender: string,
+  amount: BigNumber
+) => {
+  const { state, send: multicall } = useContractFunction(contracts[protocolName].tenderSwap, "multicall");
+  const { library } = useEthers();
+
+  const swapWithPermit = async (
+    tokenAddress: string,
+    tokenAmount: BigNumber,
+    minAmount: BigNumber,
+    deadline: number
+  ) => {
+    const permit = await signERC2612Permit(library, token, owner ?? "", spender, amount?.toString(), getDeadline());
```

Comment:
Library is not a thing here

Reply:
![image](https://user-images.githubusercontent.com/2270661/150986768-49092103-9cfb-4f4e-a92c-08bf95884a47.png)